test/utils/ShortStrings.test.js

```diff
@@ -0,0 +1,44 @@
+const { expect } = require('chai');
+const { expectRevertCustomError } = require('../helpers/customError');
+
+const ShortStrings = artifacts.require('$ShortStrings');
+
+function decode(sstr) {
+  return web3.utils.toUtf8(sstr).slice(0, parseInt(sstr.slice(64), 16));
+}
+
+contract('ShortStrings', function () {
+  before(async function () {
+    this.mock = await ShortStrings.new();
+  });
+
+  for (const str of [0, 1, 16, 31, 32, 64, 1024].map(length => 'a'.repeat(length))) {
+    describe(`with string length ${str.length}`, function () {
+      it('encode / decode', async function () {
+        if (str.length < 32) {
+          const encoded = await this.mock.$toShortString(str);
+          expect(decode(encoded)).to.be.equal(str);
+
+          const length = await this.mock.$length(encoded);
+          expect(length.toNumber()).to.be.equal(str.length);
+
+          const decoded = await this.mock.$toString(encoded);
+          expect(decoded).to.be.equal(str);
+        } else {
+          await expectRevertCustomError(this.mock.$toShortString(str), `StringTooLong("${str}")`);
+        }
+      });
+
+      it('set / get with fallback', async function () {
+        const { ret0 } = await this.mock
+          .$setWithFallback(str, 0)
+          .then(({ logs }) => logs.find(({ event }) => event == 'return$setWithFallback').args);
```

Comment:
Let's keep `then` for one-liners only, please.

```suggestion
        const { logs } = await this.mock.$setWithFallback(str, 0);
        const { ret0 } = logs.find(({ event }) => event == 'return$setWithFallback').args);
```